src/login.php

```diff
@@ -25,40 +25,191 @@
 ##########################################################################
 */
 
-global $userID, $loggedin, $_language;
+namespace webspell;
 
-$userID = 0;
-$loggedin = false;
+/**
+ * Somewhat secure login cookie (at the very least more secure than the previous
+ * implementation).
+ *
+ * This implementation generates a random key for each login and stores a
+ * hash of that key in the database. Because the keys are random, the database
+ * entries are hashed and the tokens expire after a certain amount of time,
+ * forging login cookies becomes rather hard.
+ */
+class LoginCookie
+{
+    private static function generateHash($key)
+    {
+        return hash('sha512', $key, true);
+    }
+  
+    /**
+     * Generate a cookie key. Optionally, a length in bytes can be specified,
+     * which defaults to 64.
+     * Note that on some systems, this function will not produce
+     * cryptographically strong keys due to openssl_random_pseudo_bytes not
+     * being available.
+     */
+    private static function generateKey($length = 64)
+    {
+        if (function_exists('openssl_random_pseudo_bytes')) {
+            $key = openssl_random_pseudo_bytes($length);
+        } else {
+            $key = '';
+            while ($length --> 0) {
+                $key .= pack('C', mt_rand(0, 255));
+            }
+        }
+        return $key;
+    }
+  
+    private static function setCookie($cookieName, $cookieValue, $expiration)
+    {
```

Comment:
Whoops! I've gotten too used to a compiler complaining about such things. Fixed.